db/dbformat.h

```diff
@@ -136,7 +136,11 @@ inline ValueType ExtractValueType(const Slice& internal_key) {
 
 // A comparator for internal keys that uses a specified comparator for
 // the user key portion and breaks ties by decreasing sequence number.
-class InternalKeyComparator : public Comparator {
+class InternalKeyComparator
+#ifdef NDEBUG
```

Comment:
Define a macro for this pattern?

Reply:
I prefer doing this generalization the second or third time we do it, not the first time. Does it feel OK to you?

Reply:
make sense :)